python/ray/tests/test_autoscaler.py

```diff
@@ -2717,24 +2765,40 @@ def testTerminateUnhealthyWorkers(self):
             "min_workers"
         ] = 1
         fill_in_raylet_ids(self.provider, lm)
-        autoscaler.update()
-        # Stopped node metric incremented.
-        mock_metrics.stopped_nodes.inc.assert_called_once_with()
-        # One node left.
-        self.waitForNodes(1)
 
-        # Check the node removal event is generated.
-        autoscaler.update()
-        events = autoscaler.event_summarizer.summary()
-        assert (
-            "Removing 1 nodes of type "
-            "ray-legacy-worker-node-type (lost contact with raylet)." in events
-        ), events
+        if disable_liveness_check:
+            # We've disabled the liveness check, so the unhealthy node should stick
+            # around until someone else takes care of it.
+            # Do several autoscaler updates:
+            for _ in range(10):
```

Comment:
I don't understand why there is a loop here

Reply:
Strictly speaking, it's not necessary. 

I'll add a comment to this effect:
We're demonstrating that the autoscaler will never take action to remove the unhealthy node, no matter how long it runs.

Reply:
Added the comment.